### ISSUE TITLE
Non trivial pattern for deserializing ZonedDateTime

### ISSUE THREAD

**Issue by dehann** [JuliaRobotics/DistributedFactorGraphs.jl#971]
This gives a parsing error:

```julia
using JSON3, TimeZones

struct MyType
  createdTimestamp::ZonedDateTime
end

str = "{\"createdTimestamp\":\"2020-08-11T04:05:59.82-04:00\"}"

JSON3.read(str, MyType)
```

**Comment by Affie** [comment]
That is unfortunate, it used to work out of the box. The default format for parsing is "yyyy-mm-ddTHH:MM:SS.ssszzz" so it's missing the 3rd decimal on seconds.

**Comment by Affie** [comment]
It looks like its related to this issue: https://github.com/JuliaTime/TimeZones.jl/issues/83#issuecomment-502243122

If you create the string with JSON3 it looks like it works fine. as it writes the third decimal place on seconds.

**Comment by dehann** [comment]
Think this was fixed in DFG v0.20?  If so then we can likely close this.
